src/Types/Input/EnumInputType.hack

```diff
@@ -0,0 +1,33 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Graphpinator\Parser\Value;
+use namespace Slack\GraphQL;
+
+
+abstract class EnumInputType extends NamedInputType {
+
+    <<__Enforceable>>
+    abstract const type TCoerced as arraykey;
+    abstract const \HH\enumname<this::TCoerced> HACK_ENUM;
+
+    <<__Override>>
+    public function coerceValue(mixed $value): this::TCoerced {
+        if (!$value is this::TCoerced) {
+            throw new GraphQL\UserFacingError('Expected a valid value for %s, got %s', $this::NAME, (string)$value);
+        }
+        return $value;
+    }
+
+    <<__Override>>
+    final public function coerceNonVariableNode(
+        Value\Value $node,
+        dict<string, mixed> $variable_values,
+    ): this::TCoerced {
+        $value = $node->getRawValue();
+        if (!$node is Value\EnumLiteral || !$value is this::TCoerced) {
```

Comment:
Same here.

The additional check (`!$node is Value\EnumLiteral`) is correct, we should indeed be rejecting e.g. a string literal even if its value would be a valid enum const name.

```GraphQL
some_field(enum_arg: FOO) # good
some_field(enum_arg: "FOO") # bad
```

(probably worth adding to the test)